docs/designs/schema-etl.md

```diff
@@ -0,0 +1,182 @@
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: FunctionSpec defines the desired state of Function
+            properties:
+              body:
+                description: Body is the body of the function
+                type: string
+              database:
+                description: Database is the name of the database in which the function
+                  will be created
+                type: string
+              docString:
+                description: DocString is the documentation string for the function
+                type: string
+              folder:
+                description: Folder is the folder in which the function will be created
+                type: string
+              isView:
+                description: IsView is a flag indicating whether the function is a
+                  view
+                type: boolean
+              name:
+                description: Name is the name of the function
+                type: string
+              parameters:
+                description: Parameters is a list of parameters for the function
+                items:
+                  properties:
+                    name:
+                      type: string
+                    type:
+                      type: string
+                  required:
+                  - name
+                  - type
+                  type: object
+                type: array
+              table:
+                description: |-
+                  Table is the name of the table in which the function will be created. We must
+                  specify a table if the function is a view, otherwise the Table name is optional.
+                type: string
+            required:
+            - body
+            - database
+            - name
+            type: object
+          status:
+            description: FunctionStatus defines the observed state of Function
+            properties:
+              lastTimeReconciled:
+                description: LastTimeReconciled is the last time the Function was
+                  reconciled
+                format: date-time
+                type: string
+              message:
+                description: Message is a human-readable message indicating details
+                  about the Function
+                type: string
+            required:
+            - lastTimeReconciled
+            - message
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+```
+
+A sample use is:
+
+```yaml
+apiVersion: adx-mon.azure.com/v1
+kind: Function
+metadata:
+  name: samplefn
+spec:
+  name: SampleFn
+  body: |
+    SampleFn
```

Comment:
Should/does the source table in this example need to be different than the Function name, or does it work as-is?

Reply:
They need to be the same if the desired outcome is a View that can be queried in place of a Table (facade)